### PR TITLE
Feature/auth

### LINKED ISSUE #10: Implement app authentification/authorization.

### PR DESCRIPTION
**Pull request by alina26**

### CODE REVIEW

**Review by YuliaYur — APPROVED** [review]

File: Studle.BLL/Services/UserService.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        public UserService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            this.unitOfWork = unitOfWork;
+            this.mapper = mapper;
+        }
+
+        public async Task<IdentityResult> SignUpAsync(UserDto user, string password)
+        {
+            var userEntity = mapper.Map<User>(user);
+            /*Log.Information($"GUID value is: {Guid.NewGuid().ToString()}");
```
Comment (rostyk-begey) [inline Studle.BLL/Services/UserService.cs]: Please remove unused code

File: Studle.BLL/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        public async Task<SignInResult> SignInAsync(UserDto user, string password, bool isPersistant)
+        {
+            var userEntity = unitOfWork.UserManager.FindByEmailAsync(user.Email);
+            if (userEntity.Result != null)
+            {
+                var result = await unitOfWork.SignInManager.PasswordSignInAsync(userEntity.Result, password, isPersistant, false);
+                Log.Information($"Signed in: {0} ", user);
+                return result;
+            }
+            else Log.Warning($"Email isn`t available: {0} ", user.Email);
```
Comment (rostyk-begey) [inline Studle.BLL/Services/UserService.cs]: There is no need to else statement here

File: Studle.DAL/EF/AppDbContext.cs [hunk 3]
```diff
@@ -19,13 +19,29 @@ public class AppDbContext : DbContext
...
+
+        public AppDbContext(DbContextOptions<AppDbContext> options)
+            : base(options)
+        {
+            Database.EnsureCreated();
+        }
+
+        public AppDbContext()
+        {
+            Database.EnsureDeleted();   // удаляем бд со старой схемой
```
Comment (rostyk-begey) [inline Studle.DAL/EF/AppDbContext.cs]: Please remove comments

File: Studle.WEB/Controllers/SignInController.cs [hunk 4]
```diff
@@ -1,13 +1,50 @@
...
+            if (this.ModelState.IsValid)
+            {
+                UserDto user = new UserDto { Email = model.Email };
+                var result = await this.userService.SignInAsync(user, model.Password, true);
+                if (result?.Succeeded == true)
+                {
+                    Log.Information($"Logged in: {user.Email} ");
+                    return this.Redirect("/Home/Privacy");
+                }
+                else
```
Comment (rostyk-begey) [inline Studle.WEB/Controllers/SignInController.cs]: There is no need for else statement here

File: Studle.BLL/DTO/RoleDto.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+using Studle.BLL.Dto;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Studle.BLL.Dto
+{
+    class RoleDto
```
Comment (rostyk-begey) [inline Studle.BLL/DTO/RoleDto.cs]: Please update file & class name in the same style as others

File: Studle.DAL/EF/AppDbContext.cs [hunk 6]
```diff
@@ -19,13 +19,29 @@ public class AppDbContext : DbContext
...
+            Database.EnsureCreated();
+        }
+
+
```
Comment (rostyk-begey) [inline Studle.DAL/EF/AppDbContext.cs]: Please remove extra linebreak

File: Studle.WEB/Views/SignIn/Index.cshtml [hunk 7]
```diff
@@ -1,10 +1,17 @@
...
+@{
+    ViewBag.Title = "Sign In";
+}
+
+
 @section Head {
   <link rel="stylesheet" href="~/dist/pages/sign-in.css" />
 }
 
-@section Scripts {
```
Comment (rostyk-begey) [inline Studle.WEB/Views/SignIn/Index.cshtml]: Scripts shouldn't be deleted

File: Studle.WEB/Views/SignUp/Index.cshtml [hunk 8]
```diff
@@ -1,13 +1,18 @@
...
+@{
+    ViewBag.Title = "Sign Up";
+}
+
+
 @section Head {
   <link rel="stylesheet" href="~/dist/pages/sign-up.css" />
 }
 
-@section Scripts {
```
Comment (rostyk-begey) [inline Studle.WEB/Views/SignUp/Index.cshtml]: Scripts shouldn't be deleted

File: Studle.WEB/Controllers/SignInController.cs [hunk 9]
```diff
@@ -1,13 +1,50 @@
...
+                {
+                    Log.Warning($"Log in failed {user} ");
+                }
+
+                return this.Redirect("/Home/Index");
```
Comment (rostyk-begey) [inline Studle.WEB/Controllers/SignInController.cs]: Should redirect back to SignIn page with error message shown

File: Studle.WEB/Controllers/SignUpController.cs [hunk 10]
```diff
@@ -1,13 +1,63 @@
...
+                        return this.Redirect("/SignIn/Index");
+                    }
+
+                }
+                else
+                {
+                    Log.Warning($"Registration failed {0} ", user);
+                }
+
+                return this.Redirect("/Home/Index");
```
Comment (rostyk-begey) [inline Studle.WEB/Controllers/SignUpController.cs]: Should redirect back to SignUp page with error message shown

**Review by rostyk-begey — CHANGES_REQUESTED** [review]
Please do not delete scripts from pages. Also error message have to be shown on the same page where the form have been submited from.

**Review by Kutsakh — APPROVED** [review]

**Review by rostyk-begey — APPROVED** [review]

**Pull request merged**